src/overlays/actors/ovl_Obj_Ghaka/z_obj_ghaka.c

```diff
@@ -21,26 +34,162 @@ const ActorInit Obj_Ghaka_InitVars = {
+    } else {
+        Actor_SpawnWithParentAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_BG_GORON_OYU, 0.0f, 25.0f, 261.0f, 0, 0,
+                                         0, 1, -1, this->dyna.actor.unk20, 0);
+    }
+}
+
+void func_80B3C39C(ObjGhaka* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s16 distDiff;
+
+    distDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.shape.rot.y;
+    if (func_800B84D0(&this->dyna.actor, globalCtx)) {
+        func_80B3C29C(this);
+    } else if (this->dyna.actor.xzDistToPlayer < 100.0f || this->dyna.actor.isTargeted) {
+        if (distDiff <= -0x5556 || distDiff >= 0x5556) {
+            func_800B863C(&this->dyna.actor, globalCtx);
+            if (player->unk14B == 1) {
+                this->dyna.actor.textId = 0xCF3;
+            } else {
+                this->dyna.actor.textId = 0xCF2;
+            }
+        }
+    }
+    if (this->dyna.unk148 < 0.0f && !(gSaveContext.weekEventReg[20] & 0x20) && player->unk14B == 1) {
+        func_80B3C2B0(this);
+    } else {
+        player->unkA70 &= ~0x10;
+        this->dyna.unk148 = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C29C.asm")
+void func_80B3C4E0(ObjGhaka* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C2B0.asm")
+    temp_v0 = func_80152498(&globalCtx->msgCtx);
+    if (temp_v0 == 5) {
+        if (func_80147624(globalCtx)) {
+            globalCtx->msgCtx.unk11F23 = 0x43;
+            globalCtx->msgCtx.unk12024 = 4;
+            func_80B3C260(this);
+        }
+        // Could be fake, but helps match
+    } else if ((temp_v0 & 0xFFFFFFFF) == 4) {
+        if (func_80147624(globalCtx)) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0:
+                    func_8019F208();
+                    this->dyna.actor.textId = 0xCF5;
+                    func_801518B0(globalCtx, this->dyna.actor.textId, &this->dyna.actor);
+                    break;
+                case 1:
+                    func_8019F208();
+                    this->dyna.actor.textId = 0xCF7;
+                    func_801518B0(globalCtx, this->dyna.actor.textId, &this->dyna.actor);
+                    break;
+                case 2:
+                    func_8019F230();
+                    globalCtx->msgCtx.unk11F23 = 0x43;
+                    globalCtx->msgCtx.unk12024 = 4;
+                    func_80B3C260(this);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C2C4.asm")
+void func_80B3C624(ObjGhaka* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s32 stepTemp;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C39C.asm")
+    stepTemp = Math_StepToS(&this->unk_168, 0x64, 1);
```

Comment:
```suggestion
    s32 stepTemp = Math_StepToS(&this->unk_168, 100, 1);
```

Reply:
Also, does using `u8 stepTemp` allow for the removal of `0xFF` in the if conditional?

Reply:
it does! I didn't think about that since the `0xFF` is essentially just masking 2 bytes, thanks for catching that